src/read.rs

```diff
@@ -382,12 +389,15 @@ impl<R> ZipArchive<R> {
     ) -> ZipResult<Self> {
         let initial_offset = match files.first() {
             Some((_, file)) => file.header_start,
-            None => 0,
+            None => central_start,
```

Comment:
@Pr0methean note that I thought `central_start` makes more sense here, as without files that ought to be the start of the archive.